index.js

```diff
@@ -4,141 +4,145 @@ const BridgeMethod = require("./BridgeMethod");
-    utils.verifyHashOrBlockNumber(startingBlockHashOrBlockNumber);
-
-    if (isNaN(blocksToSearch) || blocksToSearch > 100 || blocksToSearch <= 0) {
-        throw new Error('blocksToSearch must be greater than 0 or less than 100');
-    }
-
-    const startingBlockNumber = typeof startingBlockHashOrBlockNumber === 'string' && startingBlockHashOrBlockNumber.indexOf('0x') === 0 ?
-        (await web3Client.eth.getBlock(startingBlockHashOrBlockNumber)).number : startingBlockHashOrBlockNumber;
-    
-    const bridgeTxs = [];
-    for (let i = 0; i < blocksToSearch; i++) {
-        let blockNumber = parseInt(startingBlockNumber) + i;
-        let blockBridgeTxs = await getBridgeTransactionsInThisBlock(web3Client, blockNumber);
-        if (blockBridgeTxs.length) {
-            bridgeTxs.push(...blockBridgeTxs);
+    getBridgeTransactionByTxHash = async (transactionHash) => {
+        utils.verifyHashOrBlockNumber(transactionHash);
+    
+        let transaction;
+        const txReceipt = await this.web3Client.eth.getTransactionReceipt(transactionHash);
+        
+        if (txReceipt?.to === Bridge.address) {
+            const bridge = Bridge.build(this.web3Client);
+            const tx = await this.web3Client.eth.getTransaction(txReceipt.transactionHash);
+            transaction = await this.createBridgeTx(bridge, tx, txReceipt);
         }
-    }
-    return bridgeTxs;
-
-}
-
-const decodeBridgeTransaction = async (web3Client, bridgeTx, bridgeTxReceipt) => {
-    if (bridgeTx.hash !== bridgeTxReceipt.transactionHash) {
-        throw new Error(`Given bridgeTx(${bridgeTx.hash}) and bridgeTxReceipt(${bridgeTxReceipt.transactionHash}) 
-        should belong to the same transaction.`);
+        
+        return transaction;
     }
 
-    if (bridgeTxReceipt?.to !== Bridge.address) {
-        throw new Error(`Given bridgeTxReceipt is not a bridge transaction`);
+    getBridgeTransactionsInThisBlock = async (blockHashOrBlockNumber) => {
+        utils.verifyHashOrBlockNumber(blockHashOrBlockNumber);
+    
+        const block = await this.web3Client.eth.getBlock(blockHashOrBlockNumber);
+        if (!block) {
+            throw new Error(`Block ${blockHashOrBlockNumber} not found`);
+        }
+    
+        const bridgeTxs = [];
+        for (let txHash of block.transactions) {
+            const transaction = await this.getBridgeTransactionByTxHash(txHash);
+            if (transaction) {
+                bridgeTxs.push(transaction);
+            }
+        }
+        return bridgeTxs;
     }
-
-    const bridge = Bridge.build(web3Client);
-    return createBridgeTx(web3Client, bridge, bridgeTx, bridgeTxReceipt);
-}
-
-const decodeBridgeMethodParameters = (web3Client, methodName, data) => {
-    const abi = Bridge.abi.find(m => m.name === methodName);
-    if (!abi) {
-        throw new Error(`${methodName} does not exist in Bridge abi`);
+    
+    // TODO: Add test case to verify that a search going beyond the best block should fail
```

Comment:
Is it possible to tackle this TODO on this refactoring?

Reply:
I think there no need to create a new test to handle this.
Because the test 'Should fail for inexisting block number' already handles that. I manually tried to search for a block way beyond the latest block and it throw: Error: Block 123456789123456789 not found, which is exactly what the existing test is already doing. Adding an extra check to the function and then another test would be unnecessary. What do you think?